lib/ldap.c

```diff
@@ -357,7 +357,15 @@ static CURLcode ldap_do(struct Curl_easy *data, bool *done)
 #ifdef USE_WIN32_LDAP
     /* Win32 LDAP SDK doesn't support insecure mode without CA! */
     server = ldap_sslinit(host, conn->port, 1);
+#if defined(_MSC_VER) && (_MSC_VER <= 1600)
+#pragma warning(push)
+/* 'type cast' : conversion from 'int' to 'void *' of greater size */
+#pragma warning(disable:4306)
+#endif
     ldap_set_option(server, LDAP_OPT_SSL, LDAP_OPT_ON);
+#if defined(_MSC_VER) && (_MSC_VER <= 1600)
+#pragma warning(pop)
+#endif
```

Comment:
the macros could be overridden instead

~~~diff
diff --git a/lib/ldap.c b/lib/ldap.c
index 5e53f4c..dcfd3c4 100644
--- a/lib/ldap.c
+++ b/lib/ldap.c
@@ -140,6 +140,13 @@ static void _ldap_free_urldesc(LDAPURLDesc *ludp);
 #define ldap_err2string ldap_err2stringA
 #endif
 
+#ifdef USE_WIN32_LDAP
+#undef LDAP_OPT_ON
+#undef LDAP_OPT_OFF
+#define LDAP_OPT_ON                 ((void *)(size_t)1)
+#define LDAP_OPT_OFF                ((void *)(size_t)0)
+#endif
 
 static CURLcode ldap_do(struct Curl_easy *data, bool *done);
 
~~~

Reply:
This can work, though I'd prefer to guard it for the old MSVS version. This also made it clear why it's there and allowed easy removal once we drop MSVS support in the future.